src/EnergyPlus/HVACManager.cc

```diff
@@ -1105,47 +1107,46 @@ void SimHVAC(EnergyPlusData &state)
+                        AvgValue = sum(humRatInletNode) / double(DataConvergParams::ConvergLogStackDepth);
+                        if (std::abs(humRatInletNode(1) - AvgValue) >
                             DataConvergParams::HVACHumRatOscillationToler) { // last iterate differs from average
                             FoundOscillationByDuplicate = false;
                             for (StackDepth = 2; StackDepth <= DataConvergParams::ConvergLogStackDepth; ++StackDepth) {
-                                if (std::abs(state.dataConvergeParams->ZoneInletConvergence(ZoneNum).InletNode(NodeIndex).HumidityRatio(1) -
-                                             state.dataConvergeParams->ZoneInletConvergence(ZoneNum).InletNode(NodeIndex).HumidityRatio(StackDepth)) <
-                                    DataConvergParams::HVACHumRatOscillationToler) {
+                                if (std::abs(humRatInletNode(1) - humRatInletNode(StackDepth)) < DataConvergParams::HVACHumRatOscillationToler) {
                                     FoundOscillationByDuplicate = true;
                                     ShowContinueError(
                                         state,
                                         format("Node named {} shows oscillating humidity ratio across iterations with a repeated value of {:.6R}",
                                                state.dataLoopNodes->NodeID(
                                                    state.dataConvergeParams->ZoneInletConvergence(ZoneNum).InletNode(NodeIndex).NodeNum),
-                                               state.dataConvergeParams->ZoneInletConvergence(ZoneNum).InletNode(NodeIndex).HumidityRatio(1)));
+                                               humRatInletNode(1)));
                                     break;
                                 }
                             }
                             if (!FoundOscillationByDuplicate) {
+
+                                auto humRatInletNodDotProd = std::inner_product(
+                                    std::begin(ConvergLogStackARR), std::end(ConvergLogStackARR), std::begin(humRatInletNode), 0.0);
+
                                 SlopeHumRat =
-                                    (sum_ConvergLogStackARR *
-                                         sum(state.dataConvergeParams->ZoneInletConvergence(ZoneNum).InletNode(NodeIndex).HumidityRatio) -
-                                     double(DataConvergParams::ConvergLogStackDepth) *
-                                         sum((ConvergLogStackARR *
-                                              state.dataConvergeParams->ZoneInletConvergence(ZoneNum).InletNode(NodeIndex).HumidityRatio))) /
+                                    (sum_ConvergLogStackARR * sum(humRatInletNode) -
+                                     double(DataConvergParams::ConvergLogStackDepth) * humRatInletNodDotProd) /
```

Comment:
It was hard to see, but the old code was using Objexx `sum` here, but this part is just a dot product, so I pulled that out into a standalone variable here to make this easier to read. (Plus Objexx `sum` won't work with an `Array1D` and `std::array`, so I had to use an STL function any way)